pdp-server/src/api/authz/authorized_users.rs

```diff
@@ -0,0 +1,218 @@
+use crate::api::authz::forward_to_opa::send_request_to_opa;
+use crate::errors::ApiError;
+use crate::openapi::AUTHZ_TAG;
+use crate::{models::Resource, state::AppState};
+use axum::extract::State;
+use axum::{
+    http::StatusCode,
+    response::{IntoResponse, Response},
+    Json,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::Value::Object;
+use std::collections::HashMap;
+use utoipa::ToSchema;
+
+#[utoipa::path(
+    post,
+    path = "/authorized_users",
+    tag = AUTHZ_TAG,
+    request_body = AuthorizedUsersAuthorizationQuery,
+    params(
+        ("Authorization" = String, Header, description = "Authorization header"),
+    ),
+    responses(
+        (status = 200, description = "Authorized users retrieved successfully", body = AuthorizedUsersResult),
+        (status = 422, description = "Invalid request payload"),
+        (status = 500, description = "Internal server error")
+    )
+)]
+pub(super) async fn authorized_users_handler(
+    State(state): State<AppState>,
+    Json(query): Json<AuthorizedUsersAuthorizationQuery>,
+) -> Response {
+    let endpoint = if state.settings.use_new_authorized_users {
+        "/v1/data/permit/authorized_users_new/authorized_users"
+    } else {
+        "/v1/data/permit/authorized_users/authorized_users"
+    };
+    let result: serde_json::Value =
+        match send_request_to_opa::<serde_json::Value, _>(&state, endpoint, &query).await {
+            Ok(result) => result,
+            Err(err) => {
+                log::error!("Failed to send request to OPA: {}", err);
+                return ApiError::from(err).into_response();
+            }
+        };
+
+    // Attempt to extract the "result" field from the response
+    if let Object(map) = result {
```

Comment:
Perhaps convert this to a function to convert opa response to the serialized one